src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V68/Eth68ProtocolHandler.cs

```diff
@@ -0,0 +1,149 @@
+            specProvider, logManager)
+    {
+        _pooledTxsRequestor = pooledTxsRequestor;
+    }
+
+    public override void HandleMessage(ZeroPacket message)
+    {
+        switch (message.PacketType)
+        {
+            case Eth68MessageCode.NewPooledTransactionHashes:
+                NewPooledTransactionHashesMessage68 newPooledTxHashesMsg =
+                    Deserialize<NewPooledTransactionHashesMessage68>(message.Content);
+                ReportIn(newPooledTxHashesMsg);
+                Handle(newPooledTxHashesMsg);
+                break;
+            default:
+                base.HandleMessage(message);
+                break;
+        }
+    }
+
+    private void Handle(NewPooledTransactionHashesMessage68 message)
+    {
+        if (message.Hashes.Count != message.Types.Count || message.Hashes.Count != message.Sizes.Count)
+        {
+            if (Logger.IsTrace)
+                Logger.Trace($"Wrong format of {nameof(NewPooledTransactionHashesMessage68)} message. " +
+                             $"Hashes count: {message.Hashes.Count} " +
+                             $"Types count: {message.Types.Count} " +
+                             $"Sizes count: {message.Sizes.Count}");
+
+            throw new SubprotocolException($"Wrong format of {nameof(NewPooledTransactionHashesMessage68)} message. " +
+                                           $"Hashes count: {message.Hashes.Count} " +
+                                           $"Types count: {message.Types.Count} " +
+                                           $"Sizes count: {message.Sizes.Count}");
+        }
+
+        Metrics.Eth68NewPooledTransactionHashesReceived++;
+
+        Stopwatch stopwatch = Stopwatch.StartNew();
+
+        _pooledTxsRequestor.RequestTransactionsEth68(Send, message.Hashes.ToArray(), message.Sizes.ToArray(),
+            message.Types.ToArray());
+
+        stopwatch.Stop();
+
+        if (Logger.IsTrace)
+            Logger.Trace($"OUT {Counter:D5} {nameof(NewPooledTransactionHashesMessage68)} to {Node:c} " +
+                         $"in {stopwatch.Elapsed.TotalMilliseconds}ms");
+    }
+
+    public override void SendNewTransactions(IEnumerable<Transaction> txs, bool sendFullTx)
+    {
+        if (sendFullTx)
+        {
+            base.SendNewTransactions(txs, sendFullTx);
+            return;
+        }
+
+        List<TxType> types = new();
+        List<int> sizes = new();
+        List<Keccak> hashes = new();
+
+        TxDecoder txDecoder = new();
+
+        foreach (Transaction tx in txs)
+        {
+            if (hashes.Count == NewPooledTransactionHashesMessage68.MaxCount)
+            {
+                SendMessage(types, sizes, hashes);
+                types.Clear();
+                sizes.Clear();
+                hashes.Clear();
+            }
+
+            if (tx.Hash is not null)
+            {
+                types.Add(tx.Type);
+                sizes.Add(txDecoder.GetLength(tx, RlpBehaviors.None));
```

Comment:
This is a heavy operation and in this form we are calculating it for every tx, for every peer, so if we have 100 peers, we will calculate it 100x for every tx. Maybe let's do it only once and store the result? I think we should change `_persistentTxs`, `_txsToSend` and `_accumulatedTemporaryTxs` in `TxBroadcaster` to store tx size and calculate it only once, in `Broadcast`. Or even add size to `_transactions` in `TxPool` and calculate it only in `AddCore`, just before `_transactions.TryInsert`, and later pass to TxBroadcaster? @LukaszRozmej WDYT?